### ISSUE TITLE
Problem with wrong case in specifier statement

### ISSUE THREAD

**Issue by tombentley** [ceylon/ceylon-compiler#1834]
As found by @lucaswerkmeister in #1728 the following causes a backend error

```
abstract class Top() of Side | Mid {}
class Side() extends Top() {}
abstract class Mid() of Bottom extends Top() {
    shared formal Top[] children;
}
class Bottom(Mid? mid, Side[] sides) extends Mid() {
    shared actual Side[]|[Mid,Side*] children;
    if (exists mid) {
        children = [mid, *sides];
    } else {
        children = sides;
    }
}
```

> incompatible types
required: ceylon.language.Sequential<? extends tmp.Top>
found: ceylon.language.Sequence

**Comment by FroMage** [comment]
So this is fixed then I guess. Please reopen if not.

**Comment by lucaswerkmeister** [comment]
Not fixed as such, no… reported as #2026.

**Comment by FroMage** [comment]
OK, thanks
